skfda/exploratory/stats/_functional_transformers.py

```diff
@@ -75,3 +77,136 @@ def local_averages(
         for i in range(n_intervals)
     ]
     return np.asarray(integrated_data)
+
+
+def _calculate_curves_occupation_(
+    curve_y_coordinates: np.ndarray,
+    curve_x_coordinates: np.ndarray,
+    interval: Tuple[float, float],
+) -> NDArrayFloat:
+    y1, y2 = interval
+
+    if y2 < y1:
+        raise ValueError(
+            "Interval limits (a,b) should satisfy a <= b. "
+            + str(interval) + " doesn't",
+        )
+
+    # Reshape original curves so they have one dimension less
+    new_shape = curve_y_coordinates.shape[1::-1]
```

Comment:
This is difficult to understand. If you know that the last dimension is always one, you can index it using `curve[..., 0]`.